apps/functions-subscription/src/use-cases/errors.ts

```diff
@@ -0,0 +1,3 @@
+export class SubscriptionAlreadyExists extends Error {
+  readonly _tag = 'SubscriptionAlreadyExists';
+}
```

Comment:
**note**: This error is defined using class to gain access to both the error type and a free constructor